src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,17 +1,43 @@
-import React from 'react';
+/* eslint-disable max-len */
+import React, { useEffect, useState } from 'react';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as filterActions } from '../../features/filter';
+import { Select } from '../../types/helper';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const filter = useAppSelector(state => state.filter);
+  const [status, setStatus] = useState(filter.status);
+  const [query, setQuery] = useState(filter.query);
```

Comment:
If you defined `status' and `query` as parts of `filter reducer` then why you handle them as regular state? Create actions for setting `status' and `query` and trigger the dispatch